src/main/scala/com/databricks/labs/overwatch/Optimizer.scala

```diff
@@ -54,23 +54,41 @@ object Optimizer extends SparkSessionWrapper{
    */
   def main(args: Array[String]): Unit = {
 
-    val overwatchETLDB = if (args.length == 1) {
-      args(0)
+//    val overwatchETLDB = if (args.length == 2) {
+//      args(0)
+//    } else {
+//      throw new BadConfigException(s"Main class requires at least 1 but less than 5 arguments. Received ${args.length} " +
+//        s"arguments. Please review the docs to compose the input arguments appropriately.")
+//    }
+
+    val overwatchETLDB = args(0)
+    val orgId = args.lift(1).getOrElse(" ")
+
+
+    val orgIdList = spark.table(s"${overwatchETLDB}.pipeline_report").select("organization_id").distinct().collect().map(x =>x(0).toString)
+
+    val orgIDs = if (orgId != " ") {
+      if (orgIdList.contains(orgId)) {
+        Array(orgId)
+      } else {
+        throw new BadConfigException("Input Organization_ID is not part of the Overwatch Deployment for which you want to run the optimizer")
+      }
     } else {
-      throw new BadConfigException(s"Main class requires at least 1 but less than 5 arguments. Received ${args.length} " +
-        s"arguments. Please review the docs to compose the input arguments appropriately.")
+      orgIdList
     }
```

Comment:
if input orgID is given then directly call  spark.table(s"${overwatchETLDB}.pipeline_report").filter('rganization_id === args(1)) and perform a collect if it has record then we have a valid orgID.

Reply:
I have already done that but using different line. @sriram251-code  correct me if I am wrong here

Reply:
as discussed please perform the necessary code changes to remove " " and make it more efficient.